scripts/tests/test_topology.py

```diff
@@ -0,0 +1,154 @@
+from typing import Optional
```

Comment:
Are we adding running these unit tests as part of PR validation from now on ? I think we should if we don't.

Reply:
We don't yet, it will be done as a separate PR, orthogonal to this one.